packages/browser/src/tracekit.ts

```diff
@@ -70,50 +55,52 @@ export function computeStackTrace(ex: any): StackTrace {
   }
 
   try {
-    // This must be tried first because Opera 10 *destroys*
-    // its stacktrace property if you try to access the stack
-    // property first!!
-    stack = computeStackTraceFromStacktraceProp(ex);
-    if (stack) {
-      return popFrames(stack, popSize);
-    }
+    // Access and store the stacktrace property before doing ANYTHING
+    // else to it because Opera is not very good at providing it
+    // reliably in other circumstances.
+    const stacktrace = ex.stacktrace || ex.stack || '';
+
+    frames = parseFrames(stacktrace);
   } catch (e) {
     // no-empty
   }
 
-  try {
-    stack = computeStackTraceFromStackProp(ex);
-    if (stack) {
-      return popFrames(stack, popSize);
-    }
-  } catch (e) {
-    // no-empty
+  if (frames.length && popSize > 0) {
+    frames = frames.slice(popSize);
   }
 
   return {
     message: extractMessage(ex),
     name: ex && ex.name,
-    stack: [],
-    failed: true,
+    stack: frames,
   };
 }
 
 /** JSDoc */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any, complexity
-function computeStackTraceFromStackProp(ex: any): StackTrace | null {
-  if (!ex || !ex.stack) {
-    return null;
-  }
-
-  const stack = [];
-  const lines = ex.stack.split('\n');
+// eslint-disable-next-line complexity
+function parseFrames(stackString: string): StackFrame[] {
+  const frames: StackFrame[] = [];
+  const lines = stackString.split('\n');
   let isEval;
   let submatch;
   let parts;
-  let element;
+  let element: StackFrame | undefined;
 
   for (const line of lines) {
-    if ((parts = chrome.exec(line))) {
+    if ((parts = opera10Regex.exec(line))) {
+      element = {
+        filename: parts[2],
+        function: parts[3] || UNKNOWN_FUNCTION,
+        lineno: +parts[1],
```

Comment:
This is super byte size warrior, but perhaps to save on the repeated un-minifiable object props being defined when creating the `element` obj, we can extract into a function that takes args for each of these?

We do sacrifice readability of this file a little, and the savings might not justify it tbh. Thoughts?

```ts
// TS will complain about this
function createStackFrame(filename, function, lineno, colno?: StackFrame['colno']): StackFrame {
  const element = {
    filename,
    function,
    lineno,
  }

  if (colno) element.colno = colno;

  return element;
}

```

Reply:
Yes, that would mean the fields would only need to be defined once all the parsers!

I'll add that to the next PR.

Reply:
Sounds like a plan.